src/FileMigrationHelper.php

```diff
@@ -56,27 +58,40 @@ public function run($base = null)
         Environment::increaseTimeLimitTo();
         Environment::increaseMemoryLimitTo();
 
+        /** @var LoggerInterface $logger */
+        $logger = Injector::inst()->get(LoggerInterface::class);
+
         // Loop over all files
-        $count = 0;
+        $i = $migrated = 0;
         $originalState = null;
         if (class_exists(Versioned::class)) {
             $originalState = Versioned::get_reading_mode();
             Versioned::set_stage(Versioned::DRAFT);
         }
 
-        foreach ($this->getFileQuery() as $file) {
+        $query = $this->getFileQuery();
+        $total = $query->count();
+        foreach ($query as $file) {
             // Bypass the accessor and the filename from the column
             $filename = $file->getField('Filename');
 
             $success = $this->migrateFile($base, $file, $filename);
+
+            if ($i % 100 === 0) {
+                $logger->info("Iterated $i out of $total files. Migrated $migrated files.");
+            }
+
+            $i++;
             if ($success) {
-                $count++;
+                $migrated++;
+            } else {
+                $logger->warning("$filename was not migrated");
```

Comment:
Do you think it makes more sense to add this logging in the `migrateFile` task where it returns `false`? Then you can say "File was not migrated: could not locate file on the filesystem" or "File does not have a whitelisted extension"

Reply:
That would spread the logging concern around 2 methods, whereas now it seems to be contained in a single place.
I'd guess it would make sense to either move/add both - positive and negative logging in there, or just leave it here so that migrateFile doesn't contain logging concern at all.